### ISSUE TITLE
No valid solidity source files found.

### ISSUE THREAD

**Issue by MujatabaRehman** [Consensys/vscode-solidity-metrics#19]
I am trying to run solidity metrics extension on vscode but it is giving me error of '"No valid solidity source files found". Does the extension only works with truffle projects?

**Comment by piyushshukla599** [comment]
Same issue

**Comment by blockchainexpert912** [comment]
same issue.

**Comment by krypston104** [comment]
I have the same problem. No way of making it work

**Comment by Breeje16** [comment]
> Same issue

I faced the same problem, but by opening the Command Palette and selecting `Solidity Metrics: Report Metrics for all open workspaces`, it worked for me.

**Comment by 0xBoom** [comment]
> > Same issue
> 
> I faced the same problem, but by opening the Command Palette and selecting `Solidity Metrics: Report Metrics for all open workspaces`, it worked for me.

This didn't work for me but I got another error instead:

`Error loading webview: Error: Could not register service worker: InvalidStateError: Failed to register a ServiceWorker: The document is in an invalid state..`

**Comment by tintinweb** [comment]
lmk if this still happens with https://github.com/Consensys/vscode-solidity-metrics/releases/tag/v0.0.23

**Comment by Abbasjafri-syed** [comment]
yes still facing this with above version... try to install through npm but path issue problem...
'solidity-code-metrics' is not recognized as an internal or external command,
operable program or batch file.
how to add path in env variables

**Comment by Abbasjafri-syed** [comment]
> lmk if this still happens with https://github.com/Consensys/vscode-solidity-metrics/releases/tag/v0.0.23
yes still facing this with above version... try to install through npm but path issue problem...
'solidity-code-metrics' is not recognized as an internal or external command, operable program or batch file.
how to add path in env variables...
also is there a way to export the file in different format like pdf or docx rather than html

**Comment by hans-cyfrin** [comment]
I can see the issue happens only on Windows.
Running for the whole workspace works, but running from the context menu does not.
On Mac, everything works as expected. Great tool.

**Comment by technicallybean** [comment]
I'm still getting "No valid solidity source files found" on Windows 11, despite having the latest version of the extension installed (today). Does anyone know how to get it to work?

**Comment by DCVglobalnetwork** [comment]
2024 still the same error in Windows11

**Comment by DCVglobalnetwork** [comment]
![Screenshot (231)](https://github.com/Consensys/vscode-solidity-metrics/assets/105791829/08193d7e-1bd0-46a9-aceb-273fa19e585b)
![Screenshot (231)](https://github.com/Consensys/vscode-solidity-metrics/assets/105791829/85fca3c1-ac0b-49b0-b172-1c1cab8820e8)

**Comment by DCVglobalnetwork** [comment]
> I'm still getting "No valid solidity source files found" on Windows 11, despite having the latest version of the extension installed (today). Does anyone know how to get it to work?

Open the command palette (ctrl + shift + P), then type Solidity Metric: report metrics for all open workspaces click enter

**Comment by DCVglobalnetwork** [comment]
![Screenshot 2024-01-18 151638](https://github.com/Consensys/vscode-solidity-metrics/assets/105791829/3be73936-68f3-4098-b66b-3978d54d2240)

**Comment by DCVglobalnetwork** [comment]
![Uploading Screenshot 2024-01-18 151638.png…]()

**Comment by DCVglobalnetwork** [comment]
![Screenshot 2024-01-18 151638](https://github.com/Consensys/vscode-solidity-metrics/assets/105791829/90dd6055-453e-4b3c-a9ff-83f8c29088e3)

**Comment by knowAshish** [comment]
How to download solidity metrics on windows machine

**Comment by JDengin** [comment]
> How to download solidity metrics on windows machine ?

If you're using VScode you add it directly as an extension, you don't have to download it

**Comment by JDengin** [comment]
> ![Screenshot 2024-01-18 151638](https://private-user-images.githubusercontent.com/105791829/297792652-90dd6055-453e-4b3c-a9ff-83f8c29088e3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4OTkwMzgsIm5iZiI6MTcwNzg5ODczOCwicGF0aCI6Ii8xMDU3OTE4MjkvMjk3NzkyNjUyLTkwZGQ2MDU1LTQ1M2UtNGIzYy1hOWZmLTgzZjhjMjkwODhlMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxNFQwODE4NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03ZDA3MjA4ZGIxM2EwNGY1MTI2Y2UxM2QwOWQ0MWRlNTU2ODNhMDdmZjg2MGZkYWJjNzU0NWZlZmEzYjMzMWJjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.PmcbYNUUkuV_MVrqQX4J_wkg5fClR769102_sOm05vE)

This method works, thank you !

**Comment by biginfo2012** [comment]
On windows, only `Solidity Metrics: Report Metrics for all open workspaces` of Command Palette is worked.
Only on windows.

**Comment by adityaxxz** [comment]
The `Solidity Metrics` still giving `No valid solidity sources` error on Windows only. So, I booted Linux and works perfectly fine. Still can it be cured on windows?? ser @tintinweb

**Comment by abdxzi** [comment]
Still have this issue on windows 11

**Comment by nightfallxyz** [comment]
Issue persists on Windows 11.  I will say if I use the command palette fix then it will work when the single time I select 'Solidity Metrics: Report Metrics for all open workspaces' but will work if and only if I use that path to desired results.  Using solution/context menu still seems broken.

**Comment by Cohee9791** [comment]
windows 11 still have the same issue, I do not know if it is some pluging conficts or something? every sol file cannot be scoped

**Comment by Cohee9791** [comment]
Right click is not working,but the using command is fine

**Comment by Cohee9791** [comment]
> lmk if this still happens with https://github.com/Consensys/vscode-solidity-metrics/releases/tag/v0.0.23

I tried this version ,STILL CANNOT OPEN WITH RIGHT CLICK

**Comment by Zeddex** [comment]
Doesn't work no Windows 10 as well.

**Comment by vishwajeeta** [comment]
> > Same issue
> 
> I faced the same problem, but by opening the Command Palette and selecting `Solidity Metrics: Report Metrics for all open workspaces`, it worked for me.

great its working for me

**Comment by Nathan5467** [comment]
Press Ctrl + Shift + P and type Solidity Metric.
You will then be presented with four options.
Choose the one that suits you best.
